core/field_variable.js

```diff
@@ -194,6 +194,28 @@ Blockly.FieldVariable.prototype.toXml = function(fieldElement) {
   return fieldElement;
 };
 
+/**
+ * Saves this field's value.
+ * @return {string} The id of the variable referenced by this field.
+ * @override
+ * @package
+ */
+Blockly.FieldVariable.prototype.saveState = function() {
+  // Make sure the variable is initialized.
+  this.initModel();
+  return this.variable_.getId();
+};
```

Comment:
In the `toXml` function we serialize a lot more state (variable names and types). But since that info is now stored separately (and has been for a long time) I don't think it's necessary. All variables should be created before this field is deserialized from XML, so all we need is a reference to it (ie its ID).

If this is wrong, it should come up in the round-trip tests, but please check my logic! I definitely don't want to break variables haha.